src/router/index.js

```diff
@@ -0,0 +1,21 @@
+import { createRouter, createWebHistory } from "vue-router";
+import AllCoursesView from "@/views/AllCoursesView.vue";
+
+const router = createRouter({
+    history: createWebHistory(import.meta.env.BASE_URL),
+    routes: [
+        {
+            path: "/",
+            name: "all-courses",
+            component: AllCoursesView,
+        },
+        {
+            path: "/course/:courseId",
+            name: "about-course",
+            component: () => import("../views/CourseDetailsView.vue"),
+            props: true,
```

Comment:
tell please why do you use **props: true**

Reply:
I used it to pass params while using RouterLink
`RouterLink :to="{ name: 'about-course', params: {courseId: course?.id }}`

Reply:
in general, I use params during routing without field 'props'
you can test, probably `props: true,`  you can just remove

Reply:
I got what you mean but here I used another approach to pass params, and `props: true` are necessary. But the next time I'll try to avoid this